### ISSUE TITLE
Experiment 2 RSJA

### ISSUE THREAD

**Issue by jds8** [jds8/SAE4DLM-CE#20]
Rerun Experiment 1 but with random weights instead of the SAE weights. This experiment is designed to test the effect of sparsity on cross entropy loss, so we want to use untrained/random SAEs with different sparsity levels and see how that affects CE. You can create untrained SAEs simply by creating a new AutoEncoderTopK object without loading an ae.pt file. You will have to adjust the eval code to either ignore the SAEs in <DREAM_SAE_ROOT> or remove that command line argument.

For a fair comparison with the SAEs, we need to ensure that the scale of the random encoder and decoder matrices are the same as those of the SAEs. By default the encoder and decoder matrices are transposes of each other and their respective row and column norms are equal to 1. Therefore, we should compute those same norms of the trained SAE and then multiply the untrained SAE matrices by those norms:

```
untrained_sae = AutoEncoderTopK()
trained_sae = AutoEncoderTopK.from_pretrained(...)
with torch.no_grad():
    untrained_sae.encoder.weight.copy_(untrained_sae.encoder.weight * trained_sae.encoder.weight.norm(dim=1, keepdim=True))
    untrained_sae.decoder.weight.copy_(untrained_sae.decoder.weight * trained_sae.decoder.weight.norm(dim=0, keepdim=True))
```

Then use `untrained_sae` in the experiment. That's all that should be needed.

**Comment by jenniferseok** [comment]
<img width="1000" height="600" alt="Image" src="https://github.com/user-attachments/assets/43f1e482-ff02-4e57-a93a-7b6e52a854e7" />

**Comment by jds8** [comment]
Two things are interesting about this plot. 

1) Unlike with trained SAEs, in layer L1, CE loss improves as sparsity decreases (as k increases). It's not very clear what the other shallow layer, namely L5, is doing. Running with a higher token budget might make it more clear. I don't know what token budgets were used for layers other than L1, but for L1 it was 10000000.

2) It's very surprising that any of the layers, let alone all of them, actually improve CE loss with untrained SAEs (for at least one sparsity level).

**Comment by jds8** [comment]
Now we want to run this for LLaDA, so when running the eval_delta_dlm_loss.py script, use the following for the `model_name` argument: GSAI-ML/LLaDA-8B-Base. Note that LLaDA has *different SAEs* from *Dream*, so you will first have to download the new SAEs from https://huggingface.co/AwesomeInterpretability/llada-mask-topk-sae.

While the LLaDA experiments are running, work on modifying the code in train_dlm/eval_sae/eval_delta_llm_loss.py in the same way you modified the eval_delta_dlm_loss.py code (see the description at the top of this issue). We want to run the same experiment on an autoregressive LLM to see how the results compare. Once you finish updating the code, download the SAEs for Qwen: https://huggingface.co/AwesomeInterpretability/qwen-topk-sae/tree/main/saes__Qwen_Qwen2.5-7B_top_k and run with `model_name` set as Qwen/Qwen3-8B.
